messages/src/main/java/life/genny/messages/process/MessageProcessor.java

```diff
@@ -126,26 +134,46 @@ private Object processMessageHelper(QMessageGennyMSG message) {
+                }catch(ItemNotFoundException ex){
+                    log.error("Error fetching PRI_CONTEXT_ASSOCIATIONS");
+                    log.error("Exception: "+ ex.getMessage());
+                }
 
-            // Handle any default context associations
-            EntityAttribute contextAssociationsAttribute = beaUtils.getEntityAttribute(templateBe.getRealm(), templateBe.getCode(), "PRI_CONTEXT_ASSOCIATIONS");
-            String contextAssociations = contextAssociationsAttribute != null ? contextAssociationsAttribute.getValueString() : null;
-            if (contextAssociations != null) {
-                mergeUtils.addAssociatedContexts(beUtils, baseEntityContextMap, contextAssociations, false);
-            }
+                String contextAssociations = contextAssociationsAttribute != null ? contextAssociationsAttribute.getValueString() : null;
+                if (contextAssociations != null) {
+                    mergeUtils.addAssociatedContexts(beUtils, baseEntityContextMap, contextAssociations, false);
+                }
 
-            // Check for Default Message
-            if (Arrays.stream(message.getMessageTypeArr()).anyMatch(item -> item == QBaseMSGMessageType.DEFAULT)) {
-                // Use default if told to do so
-                List<String> typeList = beUtils.getBaseEntityCodeArrayFromLinkAttribute(templateBe, "PRI_DEFAULT_MSG_TYPE");
-                try {
-                    messageTypeList = typeList.stream().map(QBaseMSGMessageType::valueOf).toList();
-				} catch (Exception e) {
-					log.error(e.getLocalizedMessage());
-                    return CommonUtils.logAndReturn(log::error, e);
-				}
-            }
+                log.info("msgType: "+ Arrays.toString(message.getMessageTypeArr()));
+                
+                // Check for default msg
+                if (Arrays.stream(message.getMessageTypeArr()).anyMatch(item -> item == QBaseMSGMessageType.DEFAULT)) {
+                    log.debug("Selecting default message type");
+
+                    // Use default if told to do so
+                    List<String> typeList = null;
+                    try{
+                        typeList = beUtils.getBaseEntityCodeArrayFromLinkAttribute(templateBe, "PRI_DEFAULT_MSG_TYPE");
+                        log.debug("typeList: "+ typeList);
+                    }catch(ItemNotFoundException ex){
+                        log.error("Error fetching PRI_DEFAULT_MSG_TYPE");
+                        log.error("Exception: "+ ex.getMessage());
+                    }
+
+                    if(typeList != null){
+                        try {
+                            messageTypeList = typeList.stream().map(QBaseMSGMessageType::valueOf).toList();
+                        } catch (Exception e) {
+                            log.error(e.getLocalizedMessage());
+                            return CommonUtils.logAndReturn(log::error, e);
```

Comment:
Be careful about returning exception object. Is there a reason this method's return type is Object?

Reply:
Agreed. It is better to target specific exception types.

Reply:
done